plugins/modules/nsxt_vm_tags.py

```diff
@@ -263,17 +259,14 @@ def realize():
             for i, tag in enumerate(all_tags):
                 if TagElement(tag) in tags_to_remove:
                     all_tags[i] = None
-        persistent_tags = []
-        for tag in all_tags:
-            if tag:
-                persistent_tags += tag,
 
-        final_tags = persistent_tags
+        final_tags = [tag for tag in all_tags if tag is not None]
         final_tags_set = _get_tags_as_set(tags=final_tags)
         for tag in _read_tags_from_module_params(module.params, 'add_tags'):
-            if TagElement(tag) not in final_tags_set:
+            tag_element = TagElement(tag)
+            if tag_element not in final_tags_set:
```

Comment:
In general this kind of check should not be needed with sets, why are we doing it?